kotlin-analysis/src/main/kotlin/org/jetbrains/dokka/analysis/EnvironmentAndFacade.kt

```diff
@@ -51,7 +69,21 @@ class DokkaMessageCollector(private val logger: DokkaLogger) : MessageCollector
 }
 
 // It is not data class due to ill-defined equals
-class EnvironmentAndFacade(val environment: KotlinCoreEnvironment, val facade: DokkaResolutionFacade) {
+class EnvironmentAndFacade(
+    environment: KotlinCoreEnvironment,
+    facade: DokkaResolutionFacade,
+    private val analysisEnvironment: AnalysisEnvironment
+) : Closeable {
```

Comment:
Looks like this should now be named `EnvironmentAndFacadeAndAnalysis` :) 

What do you think about renaming it to something more universal and something that you would expect to be closeable (i.e to contain some logic) as opposed to just being a holder/pair for two  objects?

Maybe `AnalysisContext` or something?